src/test/shared/sam/debugger/awsSamDebugConfigurationValidator.test.ts

```diff
@@ -0,0 +1,136 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { instance, mock, when } from 'ts-mockito'
+
+import { CloudFormation } from '../../../../shared/cloudformation/cloudformation'
+import { CloudFormationTemplateRegistry, TemplateData } from '../../../../shared/cloudformation/templateRegistry'
+import {
+    AwsSamDebuggerConfiguration,
+    TemplateTargetProperties
+} from '../../../../shared/sam/debugger/awsSamDebugConfiguration'
+import { DefaultAwsSamDebugConfigurationValidator } from '../../../../shared/sam/debugger/awsSamDebugConfigurationValidator'
+import { createBaseTemplate } from '../../cloudformation/cloudformationTestUtils'
+
+function createTemplateConfig(): AwsSamDebuggerConfiguration {
+    return {
+        type: 'aws-sam',
+        name: 'name',
+        request: 'direct-invoke',
+        invokeTarget: {
+            target: 'template',
+            samTemplatePath: '/',
+            samTemplateResource: 'TestResource'
+        }
+    }
+}
+
+function createCodeConfig(): AwsSamDebuggerConfiguration {
+    return {
+        type: 'aws-sam',
+        name: 'name',
+        request: 'direct-invoke',
+        invokeTarget: {
+            target: 'code',
+            lambdaHandler: 'foo',
+            projectRoot: 'bar'
+        }
+    }
+}
+
+function createTemplateData(): TemplateData {
+    return {
+        path: '/',
+        template: createBaseTemplate()
+    }
+}
+
+describe('DefaultAwsSamDebugConfigurationValidator', () => {
+    const templateConfig = createTemplateConfig()
+    const codeConfig = createCodeConfig()
+    const templateData = createTemplateData()
+
+    const mockRegistry: CloudFormationTemplateRegistry = mock()
+
+    let validator: DefaultAwsSamDebugConfigurationValidator
+
+    beforeEach(() => {
+        when(mockRegistry.getRegisteredTemplate('/')).thenReturn(templateData)
```

Comment:
this doesn't need mocks. The cfnRegistry tests show how to use the registry without mocking.